### DISCUSSION TITLE
Change Language

### DISCUSSION THREAD

**Discussion by smdfritz** [tbnobody/OpenDTU#329] [Ideas]
Is it possible to introduce a drop down menu to select a different language of the web front end? (EN, DE, IT, ES,...)?

**Comment by tbnobody** [comment]
The space in the flash of devices like ESP32 is very limited. I would rather invest that in additional functionality than in translations.

**Comment by smdfritz** [comment]
OK, I understand. Thank you reply.

**Reply by trax111** [reply]
I use the Google tranlator English to German . All words tranlated are ok but   the A for Ampere is tranlated in German to the word "ein".
Could someone ste the A for Ampere to Ampere so it is not tranlated to "Ein"?

**Reply by Solarteur** [reply]
I think the google translator is "overwhelmed". "Property" --> "Eigentum", "Voltage" --> "Stromspannung" or "String--> "Saite" are examples. And then it makes no sense to use a wrong unit sign for current. So: better leave everything as it is.
![Screenshot_20221201-141857_Chrome](https://user-images.githubusercontent.com/117977338/205068027-56303c34-204e-418c-8276-f3d4dcd91b08.jpg)
![Screenshot_20221201-141830_Chrome](https://user-images.githubusercontent.com/117977338/205068050-35f6b1a0-4a04-4448-8101-f6ad7b304cd7.jpg)

**Comment by martin8310** [comment]
Hello together,

is it possible to change the words by myself? If yes, in which file? Regards Martin

**Reply by koerri** [reply]
I'd like to "bump" this. It would be great if we could at least specify the GUI language at compile time, assuming an option to switch at runtime is not feasible.

Would anyone be so kind as to point out the file(s) to be edited for this? Judging by the posts from trax111 and Solateur a manual translation should be possible.
I did poke around in the various .vue files under \OpenDTU\webapp\src\components, but any changes there didn't reflect in the web gui. And yes, my JavaScript knowledge is pretty much zero.

Thanks!

**Comment by tbnobody** [comment]
I've just pushed some commits... It took ~30kb of flash... The language (EN/DE) is based on your browser settings.

![image](https://user-images.githubusercontent.com/1753222/209412774-1d261647-7c0a-4252-9c51-7317a198723f.png)

**Reply by Solarteur** [reply]
Is it intentional that "Power DC" is untranslated?

**Reply by Archibald-Harry-Tuttle** [reply]
Gefällt Dir
Kraft GS und Kraft WS
etwa besser?

**Reply by Solarteur** [reply]
Ok, die Anführungszeichen waren vielleicht missverständlich. In den Feldern für die einzelnen Module wurde "Power" auch richtigerweise als "Leistung" übersetzt. ("Kraft" ist in diesem Zusammenhang nicht wirklich die treffende Übersetzung). DC soll natürlich so bleiben.

**Reply by tbnobody** [reply]
Ist ein bekanntes "Problem". habs hier schon geändert. geht bald raus

**Comment by trax111** [comment]
Thank you for your great work. 

 

Merry Christmas to you.

 

Thx !

 

Von: tbnobody ***@***.*** 
Gesendet: Samstag, 24. Dezember 2022 11:08
An: tbnobody/OpenDTU ***@***.***>
Cc: trax111 ***@***.***>; Comment ***@***.***>
Betreff: Re: [tbnobody/OpenDTU] Change Language (Discussion #329)

 

Ist ein bekanntes "Problem". habs hier schon geändert. geht bald raus

**Comment by MrReSc** [comment]
Ich würde die Sprache gerne wieder auf Englisch umschalten ohne das ich die Sprache im Browser anpassen muss. Kannt du in den Einstellungen eine Möglichkeit schaffen?

**Reply by koerri** [reply]
Yes, a settings option or a GUI toggle would be great.
Until then, if you want to revert to English-only, I guess you could simply delete/rename _webapp/src/locales/de.json_.

And a big Thank You @tbnobody for the the really nice work!

**Reply by MrReSc** [reply]
I think that's not how it works. You would have to rebuild the webapp.

**Reply by koerri** [reply]
@MrReSc you're right. I did do a quick test before posting but may have been mislead by some browser cache issues.
The need to rebuild the webapp would also explain my futile attempts at manual translation above.

Edit:
#473 has some information on how to get to building the webapp

**Comment by Solarteur** [comment]
Can hardly be overlooked: DE or EN can now be selected as the language for the web interface in the top line at the top right! Thanks.

**Reply by koerri** [reply]
Ist ja wie Weihnachten! :-) 
Danke ebenfalls!

**Comment by smdfritz** [comment]
Great job, thanks and happy new year.
Bzw. guten Rutsch an alle Beteiligten :-)

**Comment by MichaMEG** [comment]
Vielen lieben Dank für die sehr gute Arbeit und die sehr schnelle Weiterentwicklung auch von mir!

Ich wünsche Euch allen ein gutes neues Jahr, mit ganz viel selbst erzeugten regenerativen Strom. 

Grüße, Michael

**Comment by beneva52** [comment]
Hello there, don't know if it will be useful, but I have translated "en.json" to French.
May be a switch in the build process would suffice to select the preferred language(s) to include?
Best regards and happy new year, Ben
[fr.zip](https://github.com/tbnobody/OpenDTU/files/10320778/fr.zip)
